metron-platform/metron-writer/src/main/java/org/apache/metron/writer/kafka/KafkaWriter.java

```diff
@@ -197,6 +209,15 @@ public void init(Map stormConf, TopologyContext topologyContext, WriterConfigura
     return producerConfig;
   }
 
+  public Optional<String> getKafkaTopic(JSONObject message) {
+    if(kafkaTopicField != null) {
+      return Optional.ofNullable((String)message.get(kafkaTopicField));
+    }
+    else {
+      return Optional.ofNullable(kafkaTopic);
+    }
```

Comment:
It would make sense to add some debug statements showing which topic was chosen and what the field value is.  This would make debugging a routing issue much simpler for the user.